src/cli/cli_dataset.cpp

```diff
@@ -622,6 +622,47 @@ otError Dataset::Print(otOperationalDatasetTlvs &aDatasetTlvs)
     return error;
 }
 
+otError Dataset::PrintNonsensitive(otOperationalDatasetTlvs &aDatasetTlvs)
```

Comment:
This duplicates quite some code. What about adding a parameter to `Dataset::Print()` to specify whether it should print sensitive fields?

Reply:
Done.